EIPS/eip-5982.md

```diff
@@ -0,0 +1,92 @@
+---
+eip: 5982
+title: Role-based Access Control
+description: An interface for role-based access control for smart contracts.
+author: Zainan Victor Zhou (@xinbenlv)
+discussions-to: https://ethereum-magicians.org/t/eip-5982-role-based-access-control/11759
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-11-15
+requires: 165, 5750
+---
+
+## Abstract
+
+This EIP defines an interface for role-based access control for smart contracts. Roles are defined as `byte32`. The interface specifies how to read, grant, create and destroy roles. It specifies the sense of role power in the format of its ability to call a given method
```

Comment:
```suggestion
This EIP defines an interface for role-based access control for smart contracts. Roles are the `keccak256` hash of the human-readable role name (in SCREAMING_SNAKE_CASE). The interface specifies how to read, grant, create and destroy roles.
```
I don't understand the last sentence. Mind explaining it?

Reply:
> Roles are the `keccak256` hash of the human-readable role name (in SCREAMING_SNAKE_CASE).

While it's recommend to use `keccak256` upon a human readable string, this EIP _doesn't intend to mandate_ the choice of hash function. So I would probably not include this suggested update.

>> It specifies the sense of role power in the format of its ability to call a given method
>  I don't understand the last sentence. Mind explaining it?

I will revise that last sentence, it means this EIP not only define a way to get and set roles, but also defines a way to get the "power" of a role. The way to check if a role has certain power is by calling `canExecute` function and pass in a `bytes4` method selector in its second parameter

```
function canExecute(bytes32 executor, bytes4 method, bytes32 calldata payload, bytes calldata _data) view external returns(bool);
```